hail/src/main/scala/is/hail/expr/types/physical/PContainer.scala

```diff
@@ -192,7 +193,53 @@ abstract class PContainer extends PIterable {
     else
       Code(
         Region.storeInt(aoff, length),
-        Region.setMemory(aoff + const(4), nMissingBytes(length), const(if (setMissing) (-1).toByte else 0.toByte)))
+        Region.setMemory(aoff + const(4), nMissingBytes(length).toL, const(if (setMissing) (-1).toByte else 0.toByte)))
+  }
+
+  def zeroes(region: Region, length: Int): Long = {
+    val aoff = region.allocate(contentsAlignment, contentsByteSize(length))
```

Comment:
call the `allocate` method on PContainer

Reply:
also, should this check that the element type is numeric? Somewhat meaningless otherwise.